e2e_tests/tests/cluster/test_ship_logs.py

```diff
@@ -290,15 +299,10 @@ def handle_sigint(*arg):
     @pytest.mark.e2e_cpu
     def test_exit_wait_time_is_honored(self) -> None:
         cmd = mkcmd("print('hello world')")
-        # We need a misbehaving server to guarantee the shipper times out.
-        # This misbehaving server will listen without ever accepting.
-        with socket.socket() as listener:
-            listener.bind(("127.0.0.1", 0))
-            listener.listen(10)
-            _, port = listener.getsockname()
-            master_url = f"http://127.0.0.1:{port}"
+        # We configure the server to reject logs in order to guarantee the shipper times out.
+        with ShipLogServer(reject_logs=True) as srv:
```

Comment:
This is nice. For me, the reader, was the test broken before?

Reply:
Before, we used a completely dysfunctional server and that caused the POST logs request to fail.

Now, we have to respond to the one probe request validly, then we have to break for all the subsequent POST logs requests.

So I'm not entirely sure how to parse "before" in your question, but I can make the following factual statements that will probably answer whichever question you are asking:
- it was broken after changing ship_logs.py and before making this change, as in "this change is a necessary part of this PR"
- it was working on main